### ISSUE TITLE
Recipe Compile Error

### ISSUE THREAD

**Issue by dejuid00** [Graylog2/graylog2-server#939]
Running the 1.0.0-rc.1 (56f861f) (Moose) VMware OVA, I had only run reconfigure as I'd changed the SMTP port (previously, all changes were just followed by reboots). After this failure, I rebooted and everything still seems fine.
```
ubuntu@graylog-beta:/opt/graylog/conf$ sudo graylog-ctl reconfigure
Starting Chef Client, version 12.0.3
Compiling Cookbooks...
Recipe: graylog::default
  * directory[/etc/graylog] action create (up to date)
[2015-02-03T13:56:11-05:00] WARN: IP change detected!

  
  ================================================================================
  Recipe Compile Error in /opt/graylog/embedded/cookbooks/graylog/recipes/default.rb
  ================================================================================
  
  Net::HTTPFatalError
  -------------------
  500 "Internal Server Error"
  
  Cookbook Trace:
  ---------------
    /opt/graylog/embedded/cookbooks/graylog/libraries/graylog.rb:151:in `generate_settings'
    /opt/graylog/embedded/cookbooks/graylog/libraries/graylog.rb:198:in `generate_config'
    /opt/graylog/embedded/cookbooks/graylog/recipes/default.rb:13:in `from_file'
  
  Relevant File Content:
  ----------------------
  /opt/graylog/embedded/cookbooks/graylog/libraries/graylog.rb:
  
  144:        Graylog['local_connect'] = false if Graylog['local_connect'].nil?
  145:        Graylog['current_address'] = Graylog[:node][:ipaddress]
  146:        Graylog['last_address']  ||= Graylog['current_address']
  147:  
  148:        if Graylog['current_address'] != Graylog['last_address']
  149:          Chef::Log.warn("IP change detected!")
  150:          client = Etcd.client(host: Graylog['master_node'], port: 4001)
  151>>         client.delete("/servers/#{Graylog['last_address']}")
  152:          client.delete("/elasticsearch/#{Graylog['last_address']}")
  153:          Graylog['last_address'] = Graylog['current_address']
  154:        end
  155:  
  156:        if File.directory?("/etc/graylog")
  157:          File.open("/etc/graylog/graylog-settings.json", "w") do |f|
  158:            f.puts(
  159:              Chef::JSONCompat.to_json_pretty({
  160:                'timezone' => Graylog['timezone'],
  
  
  Running handlers:
[2015-02-03T14:01:11-05:00] ERROR: Running exception handlers
  Running handlers complete
[2015-02-03T14:01:11-05:00] ERROR: Exception handlers complete
[2015-02-03T14:01:11-05:00] FATAL: Stacktrace dumped to /opt/graylog/embedded/cookbooks/cache/chef-stacktrace.out
  Chef Client failed. 0 resources updated in 301.929902257 seconds
[2015-02-03T14:01:11-05:00] ERROR: 500 "Internal Server Error"
[2015-02-03T14:01:11-05:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
ubuntu@graylog-beta:/opt/graylog/conf$ 
```

**Comment by mariussturm** [comment]
Thanks for the report!
Could you please provide the following for further debugging:

  * all and exact comannds you issued till this error appeared
  * output of `ifconfig` after every reboot
  * output of `cat /etc/graylog/graylog-settings.json` after every `graylog-ctl reconfigure`

**Comment by dejuid00** [comment]
Hi. During the configuration, I've rebooted many times. All commands were sudo'd.

Generally, I:
* config'd the /etc/network/interfaces file
* used graylog-ctl set-email-config and graylog-ctl set-timezone - which I discovered were unsuccessful in the effect I expected post reboot
* ran dpkg-reconfigure tzdata to set TZ
* created syslog UDP input on 514, but this didn't process ESXi syslog correctly (though it looks like this was meant to be fixed in v 0.20?)
* config'd rsyslogd to listen on UDP port 514 and set to forward to Graylog2 on 5514 (this was necessary as Graylog2 still doesn't understand ESXi syslog)
* created syslog UDP input on 5514
* config'd graylog.conf for email transport
This is where I attempted graylog-ctl reconfigure and got the error, so rebooted.

ubuntu@graylog-beta:/opt/graylog/conf$ ifconfig
eth0      Link encap:Ethernet  HWaddr 00:50:56:8b:10:01  
          inet addr:10.255.140.20  Bcast:10.255.143.255  Mask:255.255.252.0
          inet6 addr: fe80::250:56ff:fe8b:1001/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:4929239 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3223 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1099920144 (1.0 GB)  TX bytes:1231596 (1.2 MB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:15645540 errors:0 dropped:0 overruns:0 frame:0
          TX packets:15645540 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:3841351432 (3.8 GB)  TX bytes:3841351432 (3.8 GB)

ubuntu@graylog-beta:/opt/graylog/conf$ cat /etc/graylog/graylog-settings.json
{
  "timezone": "America/New_York",
  "smtp_server": "smtpout.donovandata.com",
  "smtp_port": 587,
  "smtp_user": "",
  "smtp_password": "",
  "master_node": "127.0.0.1",
  "local_connect": false,
  "current_address": "127.0.0.1",
  "last_address": "127.0.0.1"
}ubuntu@graylog-beta:/opt/graylog/conf$ 

Hope that helps.

**Comment by mariussturm** [comment]
Thanks for the details!
It is not so easy to really debug this case via remote but I suspect that at some point your network interface eth0 was down and you ran `graylog-ctl reconfigure`. That broke the configuration.

I have added a check to the script to prevent this and exit if there is no proper IP set.

To get you running I would suggest that you re-download the OVA and start a fresh appliance, afterwards:

  * Login
  * edit `/etc/network/interfaces`
  * `sudo ifdown eth0 && sudo ifup eth0`
  * run `sudo graylog-ctl reconfigure`
  * wait until you can login to the web interface
  * configure Graylog `sudo graylog-ctl set-email-config` and `sudo graylog-ctl set-timezone`
  * configure the system timezone `sudo dpkg-reconfigure tzdata`
  * run `graylog-ctl reconfigure`

Everything should be configured and running at this point. There is no need to restart the appliance. This creates more problems because you possible get another IP from the DHCP server.

I close the ticket for now, there is nothing I can do on the software site. If you still see problems please ask again on the mailing list. We can have a longer conversation there.
